File: playwright-python/pages/landing_page.py
from playwright.sync_api import Page

from config import BASE_URL


class LandingPage:
    def __init__(self, page: Page):
        self.page = page
        self.title = page.get_by_role(
            "heading", name="Test thoughtfully. Build confidently."
        )
        self.start_button = page.get_by_role("link", name="Start Evaluation")

    def goto(self):
        self.page.goto(BASE_URL)

    def start_evaluation(self):
        self.start_button.click()

File: playwright-typescript/pages/LandingPage.ts
import { Page } from '@playwright/test';

export class LandingPage {
  constructor(private readonly page: Page) {}

  readonly title = () => this.page.getByRole('heading', { name: 'Test thoughtfully. Build confidently.' });
  readonly startButton = () => this.page.getByRole('link', { name: /Start Evaluation/ });

  async goto() { await this.page.goto('/'); }
  async startEvaluation() { await this.startButton().click(); }
}
